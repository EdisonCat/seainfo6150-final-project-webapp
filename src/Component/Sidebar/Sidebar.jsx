import React, {useContext} from "react";
import { Link } from "react-router-dom";
import styles from "./Sidebar.module.css";
import MarvelFandomContext from "../../Context/MarvelFandomContext";
export default function Sidebar(props) {
    const context = useContext(MarvelFandomContext);
    const handleLogout = () => {
        context.logout();
        props.setSidebar(false);
    }
	return (
		<div
			className={styles.container}
			style={props.sidebar ? { left: "0px", top: "113px" } : {}}
		>
			<Link
				className={styles.link}
				to="/"
				onClick={() => props.setSidebar(false)}
			>
				Home
			</Link>
			{props.loggedIn ? (
				<Link className={styles.link} to="/" onClick={() => handleLogout()}>
					Log Out
				</Link>
			) : (
				<>
					<Link
						className={styles.link}
						to="/signup"
						onClick={() => props.setSidebar(false)}
					>
						Sign Up
					</Link>
					<Link
						className={styles.link}
						to="/login"
						onClick={() => props.setSidebar(false)}
					>
						Log In
					</Link>
				</>
			)}
			<Link
				className={styles.link}
				to="/characters"
				onClick={() => props.setSidebar(false)}
			>
				Characters
			</Link>
			<Link
				className={styles.link}
				to="/comics"
				onClick={() => props.setSidebar(false)}
			>
				Comics
			</Link>
		</div>
	);
}
