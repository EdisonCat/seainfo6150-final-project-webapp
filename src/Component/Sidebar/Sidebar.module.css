@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@100&family=Zen+Dots&display=swap');
.container {
    background-color: rgba(22, 22, 22, 0.8);
    transition: 0.5s;
    z-index: 1;
    display: flex;
    flex-direction: column;
    align-items: center;
    color: white;
    position: absolute;
    left: -100%;
    top: 113px;
    width: 100%;
    height: 100vh;
	font-family: 'Montserrat', sans-serif;
}
.link {
    text-decoration: underline;
    text-decoration-color: transparent;
    outline: none;
    color: white;
    width: 100%;
    text-align: center;
    margin: 20px;
    transition: 0.5s;
}
.link:hover {
    text-decoration-color: red;
}