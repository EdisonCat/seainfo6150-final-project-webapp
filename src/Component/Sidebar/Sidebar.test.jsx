import React from "react";
import Sidebar from "./Sidebar";
it("rendered sidebar correctly", () => {
	// eslint-disable-next-line no-undef
	const { container } = render(<Sidebar />);
	expect(container).toMatchSnapshot();
});
