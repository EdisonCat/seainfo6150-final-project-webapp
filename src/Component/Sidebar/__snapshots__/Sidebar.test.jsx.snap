// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendered sidebar correctly 1`] = `
<div>
  <div
    class="container"
  >
    <a
      class="link"
      href="/"
    >
      Home
    </a>
    <a
      class="link"
      href="/signup"
    >
      Sign Up
    </a>
    <a
      class="link"
      href="/login"
    >
      Log In
    </a>
    <a
      class="link"
      href="/characters"
    >
      Characters
    </a>
    <a
      class="link"
      href="/comics"
    >
      Comics
    </a>
  </div>
</div>
`;
